File: financeTracking/src/main/java/com/personal/financeTracking/account/entities/Account.java
package com.personal.financeTracking.account.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.personal.financeTracking.enums.AccountType;
import com.personal.financeTracking.user.entities.User;
import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "tb_accounts")
@EntityListeners(AuditingEntityListener.class)
public class Account {


    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String bankName;

    @Column(nullable = false, unique = true)
    private String accountNumber;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    @Column(nullable = false)
    private Double balance;

    private boolean active = true;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    private User user;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: financeTrackingGo/domain/account.go
package domain

type Account struct {
	ID           string  `json:"id"`
	ACCOUNT_TYPE string  `json:"account_type"`
	ACTIVE       bool    `json:"active"`
	BALANCE      float64 `json:"balance"`
	BANK_NAME    string  `json:"bank_name"`
	CREATED_AT   string  `json:"created_at"`
	UPDATED_AT   string  `json:"updated_at"`
	USER_ID      string  `json:"user_id"`
}

type AccountService interface {
	Post(account *Account) (*Account, error)
	Put(Account *Account) error
	Delete(id string) error
	GetByID(id string) (*Account, error)
}

type AccountStorage interface {
	Insert(account *Account) (*Account, error)
	Update(Account *Account) error
	Delete(id string) error
	FindByID(id string) (*Account, error)
}

func NewAccount(
	id string,
	accountType string,
	active bool,
	balance float64,
	bankName string,
	createdAt string,
	updatedAt string,
	userID string,
) *Account {
	return &Account{
		ID:           id,
		ACCOUNT_TYPE: accountType,
		ACTIVE:       active,
		BALANCE:      balance,
		BANK_NAME:    bankName,
		CREATED_AT:   createdAt,
		UPDATED_AT:   updatedAt,
		USER_ID:      userID,
	}
}
